src/components/Elements/Map/MapScreen.tsx

```diff
@@ -324,6 +324,7 @@ const MapScreen = (): JSX.Element => {
         // bottomSheetRef.current?.snapToIndex(1)
         _removeAllGeoJson(mapRef)
         _addGeoJson()
+        _injectMarker(mapRef, mapCenter, colors)
```

Comment:
Using the parameter mapCenter the newly injected marker gets placed at the default center instead of at the previous selected room position. Since the room center coords are not available here, I suggest creating a new leaflet function that updates just the marker icon wihtout replacing the whole marker.
`window.marker.setIcon(L.icon({}))`